blessed/sequences.py

```diff
@@ -38,15 +68,36 @@ def _sort_sequences(regex_seqlist):
     # typically occur for output sequences, though with so many
     # programmatically generated regular expressions for so many terminal
     # types, it is feasible.
-    # pylint: disable=bad-builtin
-    #         Used builtin function 'filter'
-    return sorted(list(filter(None, regex_seqlist)), key=len, reverse=True)
+    return sorted((x for x in regex_seqlist if x and x[1]),
+                  key=lambda x: len(x[1]),
+                  reverse=True)
+
+
+def _unique_names(regex_seqlist):
+    """
+    Return a list of (name, pattern) pairs liki input but with names unique.
+
+    By adding _1, _2, _3 etc. to the name of each name_pair sequence the
+    first element of each tuple
+
+    :arg list regex_seqlist: list of tuples of (str name, str pattern)
+    :rtype: list
+    :returns: a copy of the input list with names modified to be unique
+    """
+    counter = collections.Counter()
+
+    def inc_get(key):
+        counter[key] += 1
+        return counter[key]
+
+    return [(u"{}_{}".format(name, inc_get(name)), pattern)
+            for name, pattern in regex_seqlist]
 
 
 def _build_numeric_capability(term, cap, optional=False,
                               base_num=99, nparams=1):
     r"""
-    Return regular expression for capabilities containing specified digits.
+    Return regular expressions for capabilities containing specified digits.
```

Comment:
should be changed back to singular "expression", and it's now a tuple of name and regex